standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/PartitionManagementTask.java

```diff
@@ -137,7 +138,8 @@ public void run() {
           try {
             Table table = msc.getTable(tableMeta.getCatName(), tableMeta.getDbName(), tableMeta.getTableName());
             Database db = msc.getDatabase(table.getCatName(), table.getDbName());
-            if (partitionDiscoveryEnabled(table.getParameters()) && !isTargetOfReplication(db)) {
+            if (partitionDiscoveryEnabled(table.getParameters()) && !isTargetOfReplication(db)
```

Comment:
For any of the tables of a db which is a target of replication, we may not need msc.getTable call

Reply:
Changes already done.